pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -55,16 +62,95 @@ type objectCacheItem struct {
 func (i *objectCacheItem) stop() bool {
 	i.lock.Lock()
 	defer i.lock.Unlock()
+	return i.stopWithoutLock()
+}
+
+func (i *objectCacheItem) stopWithoutLock() bool {
 	select {
 	case <-i.stopCh:
 		// This means that channel is already closed.
 		return false
 	default:
 		close(i.stopCh)
+		i.stopped = true
```

Comment:
If you introduce the "stopped" field, you no longer need a select here.

Basically this function can be simplified to:
```
if i.stopped {
  return false
}
i.stopped = true
close(i.stopCh)
...